### PR TITLE
feature/add-service-based-code-generation-support

### PR DESCRIPTION
**Pull request by patrick-tolosa**
This PR adds Service based code generation support.

Contents
1. CodeGen - Add config.js file with required configuration.
2. Tests - Use the sanity runner package to create local test files
3. Fix - Add missing support for assetType on a CloudinaryAsset

### CODE REVIEW

File: devTools/package.json [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+  "version": "1.0.0",
+  "description": "",
+  "main": "index.js",
+  "scripts": {
+    "start": "ts-node index.ts"
+  },
+  "author": "",
+  "license": "ISC",
+  "dependencies": {
+    "sdk-sanity-generator": "git+ssh://git@github.com/CloudinaryLtd/sdk-sanity-generator.git",
```
Comment (const-cloudinary) [inline devTools/package.json:12]: Keeping links to github.com/CloudinaryLtd in a public repo is kind of weird. nobody except us can run it... maybe we should move sdk-sanity-generator repo to cloudinary?

File: .gitignore [hunk 2]
```diff
@@ -101,3 +101,9 @@ test-output/
 ### AppEngine template
 # Google App Engine generated folder
 appengine-generated/
+
+# Compilation tests
+url-gen/src/test/kotlin/com/cloudinary/CompilationTest.kt
+devTools/error.json
+devTools/success.json
+devTools/results.json
```
Comment (const-cloudinary) [inline .gitignore:109]: Please set up your IDE to put new line at the end of file on save. These red signs really distract. thanks!

File: .code-generation/config.js [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+    groupDelimiter: '.',
+    hideActionGroups: false,
+    specialCharacterMapping: [
+      ['$', '\$']
+    ],
+    openQualifiersChar : ' {',
+    closeQualifiersChar : ' }',
+    openActionChar: '(',
+    closeActionChar: ')',
+    unsupportedTxParams: ['fl_waveform', 'fl_animated', 'e_tint', 'u_', 'e_theme', 'l_fetch', 'l_text', 'u_text', 'af_', 'l_video:', 'if_', 'e_fade', 'c_fill', 'palette_f00', 'g_ocr_text'],
```
Comment (const-cloudinary) [inline .code-generation/config.js:41]: That's a lot! 
no underlay?

**Review by const-cloudinary — APPROVED** [review]
See a few small comments, you can merge anyway.

File: devTools/package.json (hunk 1 above)
Comment (patrick-tolosa) [reply devTools/package.json:12]: I thought as well about making the library public.
I think there's no strong reason against it.

To be discussed.

File: .gitignore (hunk 2 above)
Comment (patrick-tolosa) [reply .gitignore:109]: I'm strongly against new lines in end of files 😄 😄

File: .code-generation/config.js (hunk 3 above)
Comment (patrick-tolosa) [reply .code-generation/config.js:41]: Same problem with overlay,
We don't have the correct source.format() syntax, this is partially due to the code generator.
I think once we stabilise the code generator and finish aligning the spec, we should revisit this list.

**Pull request merged**